frontend/projects/abtesting/src/app/features/dashboard/home/components/experiment-post-condition/experiment-post-condition.component.ts

```diff
@@ -22,26 +26,34 @@ export class ExperimentPostConditionComponent implements OnInit, OnChanges {
   experimentConditions = [
     { value: 'default', id: 'default' }
   ];
-  constructor(private _formBuilder: FormBuilder) { }
+  newExperimentStatSub: Subscription;
+
+  constructor(private experimentService: ExperimentService,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private _formBuilder: FormBuilder) { }
 
   ngOnChanges() {
-    this.experimentConditions = [
-      { value: 'default', id: 'default' }
-    ];
-    if (this.newExperimentData.conditions && this.newExperimentData.conditions.length) {
-      this.newExperimentData.conditions.map(value => {
-        let isConditionExist;
-        if (value.id) {
-          isConditionExist = this.experimentConditions.find((condition) => condition.id === value.id);
-          
-        } else {
-          isConditionExist = true;
-        }
-        this.experimentConditions = isConditionExist
-          ? this.experimentConditions
-          : [ ...this.experimentConditions, { value: value.conditionCode, id: value.id }];
-      });
-    }
+      this.experimentConditions = [
+        { value: 'default', id: 'default' }
+      ];
+      if (this.experimentInfo) {
+        this.newExperimentStatSub = this.experimentService.selectExperimentById(this.experimentInfo.id).subscribe(stat => {
```

Comment:
Can you please update the variable name from the `this.newExperimentStatSub` to `this.experimentSub` as the selector is returning the experiment metadata.

Also the variable from `state` to experimentData` may be?

Reply:
done